src/shared/components/query/QueryAndDownloadTabs.tsx

```diff
@@ -14,21 +15,24 @@ const QUERY = 'query';
 
 interface IQueryAndDownloadTabsProps
 {
-	store:QueryStore;
 	onSubmit?:()=>void;
 	showDownloadTab?:boolean;
+    getQueryStore:()=>QueryStore;
+    showAlerts?:boolean;
 }
 
 
 @observer
 export default class QueryAndDownloadTabs extends React.Component<IQueryAndDownloadTabsProps, {}>
 {
 
-	get store()
-	{
-		return this.props.store;
+	constructor(props:IQueryAndDownloadTabsProps){
+		super();
+		this.store = props.getQueryStore();
 	}
 
+	@observable.ref store: QueryStore;
```

Comment:
you only want to set this on init?

Reply:
the basic idea is that when i show the the query component, i want to create a store for that session/lifecycle of the component.    it cannot be replaced/updated by anything.  and it gets throw out on close.   i could model this above in the page, but this seems cleaner to me.  let me know what you think.